File: UnityMMOClient/Assets/Game/Net/Script/Net/CsProtocol/Protocol/PhysicalPower/PhysicalPowerUserPropsRequest.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Net.CsProtocol.Buffer;

namespace ZJYFrameWork.Net.CsProtocol.Protocol.PhysicalPower
{
    /// <summary>
    /// 使用体力
    /// </summary>
    public class PhysicalPowerUserPropsRequest : Model, IPacket
    {
        /// <summary>
        /// 使用体力的数量
        /// </summary>
        public int usePropNum { get; set; }

        public short ProtocolId()
        {
            return 1025;
        }

        public static PhysicalPowerUserPropsRequest ValueOf(int usePropNum)
        {
            var packet = new PhysicalPowerUserPropsRequest
            {
                usePropNum = usePropNum
            };
            return packet;
        }
    }

    public class PhysicalPowerUserPropsRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1025;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                return;
            }

            var packetData = (PhysicalPowerUserPropsRequest)packet;
            var message = new ServerMessageWrite(packetData.ProtocolId(), packetData);
            var json = JsonConvert.SerializeObject(message);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            return null;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/physicalPower/physicalPowerUserPropsRequest.lua
---@class PhysicalPowerUserPropsRequest
local PhysicalPowerUserPropsRequest = class("PhysicalPowerUserPropsRequest")
function PhysicalPowerUserPropsRequest:ctor()
    --- 使用体力会被扣除
    ---@type number
    self.usePropNum = 0
end

---@param usePropNum number 使用体力会被扣除
---@return PhysicalPowerUserPropsRequest
function PhysicalPowerUserPropsRequest:new(usePropNum)
    self.usePropNum = usePropNum --- int
    return self
end

---@return number
function PhysicalPowerUserPropsRequest:protocolId()
    return 1025
end

---@return string
function PhysicalPowerUserPropsRequest:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            usePropNum = self.usePropNum
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return PhysicalPowerUserPropsRequest
function PhysicalPowerUserPropsRequest:read(data)

    local packet = self:new(
            data.usePropNum)
    return packet
end

--- 使用体力会被扣除
---@return number 使用体力会被扣除
function PhysicalPowerUserPropsRequest:getUsePropNum()
    return self.usePropNum
end


return PhysicalPowerUserPropsRequest
